### PR TITLE
In release process merge master back into develop

### PR DESCRIPTION
**Pull request by mattias-p**
## Purpose

This PR makes `git describe --tags origin/develop` find the latest release tag. This is very helpful when you're marking up dist files during release testing.

## Context

N/A

## Changes

Adds another step to the release process.

## How to test this PR

Running the following commands, notice how the first `describe` command references an older release and counts a greater number of subsequent commits while the second `describe` command references the latest release and counts a lesser number of subsequent commits.

```sh
git co -b test origin/develop
git describe --tags
git merge origin/master -m "Merge master back into develop"
git describe --tags
```

### CODE REVIEW

File: docs/internal-documentation/maintenance/ReleaseProcess-release.md
Comment (matsduf) [inline docs/internal-documentation/maintenance/ReleaseProcess-release.md:303]: 1. Is this new PR to be created after "`merge-develop-into-master` into `master`" has been merged, or can it be created before? That should be clarified.

2. This new text is at the end of a section with the header "Merge develop branch into master", and the section is really about that. I suggest that the new section is moved to a new section 17.

File: docs/internal-documentation/maintenance/ReleaseProcess-release.md
Comment (matsduf) [inline docs/internal-documentation/maintenance/ReleaseProcess-release.md:305]: Independent, this sentence (paragraph) should come before the new text.

**Review by matsduf — CHANGES_REQUESTED** [review]
This adds another step. Can you clarify what we gain?

File: docs/internal-documentation/maintenance/ReleaseProcess-release.md
Comment (matsduf) [inline docs/internal-documentation/maintenance/ReleaseProcess-release.md:305]: It is a long section, and the second merge can easily missed. Secondly, it should be clarified that you must wait for the first merge before you can create the second.

Please update with
```
### 16.2 Merge master into develop

When the previous pull request has been merged, create a pull request from `master` on github back
into `develop` and have it merged through the normal process.
```

On line 282 add
```
### 16.1 Merge develop into master
```

**Review by matsduf — CHANGES_REQUESTED** [review]
The release process is already complex. Your proposal adds an extra step in the middle of the process. I am not against the step, but put at the end, as section 20.

It is really not part of the release, but a preparation for work after release. Though, it should be done first, before other PRs are merged to the develop branch

**Comment by mattias-p** [comment]
> It is really not part of the release, but a preparation for work after release.

Yeah. Unless we define it to be part of the release.

I'll add the extra headings you're requesting.

**Comment by mattias-p** [comment]
> Nice suggestion ! This would improve how our git tree behaves.
> 
> A question on the `merge-develop-into-master` branch (I think we already discussed this some time ago, I can't remember the conclusion sorry). What is the purpose of this step? Can't we directly merge develop into master without an extra branch?

The reason is that we want to merge the contents of the develop branch into the master branch without having to resolve any conflicts. What's causing the conflicts is when we accidentally merge something into the master branch instead of the develop branch.

The way I see thing, the reason we're having this entire problem is because we're using the master branch as a publication platform for our documentation. The developer experience would be much nicer if we had another publication platform for our documentation.

**Comment by mattias-p** [comment]
@matsduf Please review again.

**Review by matsduf — APPROVED** [review]
In another channel you raised the question if "the normal process", i.e. PR approval, is needed for the merging in section 16 and 20. If the process is followed correctly such an approval does not really add anything, but it can delay the process.

File: docs/internal-documentation/maintenance/ReleaseProcess-release.md
Comment (pnax) [inline docs/internal-documentation/maintenance/ReleaseProcess-release.md:269]: Since there is a new section, do we still need to rename this section?

**Review by pnax — APPROVED** [review]

**Pull request merged**
